### PR TITLE
Database setup

### PR DESCRIPTION
**Pull request by teelu**
Worked together to set up database and models

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: bundle exec rails s
```
Comment (h8rry) [inline Procfile]: `web: bundle exec rails server -p $PORT`

File: app/models/valet.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Valet < ActiveRecord::Base
+  # Include default devise modules.
+  devise :database_authenticatable, :registerable,
+          :recoverable, :rememberable, :trackable, :validatable,
+          :confirmable, :omniauthable
+  include DeviseTokenAuth::Concerns::User
+
+  has_many :logs
```
Comment (h8rry) [inline app/models/valet.rb]: `has_many :valet_logs`?

File: app/models/request.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Request < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :valet_pick_up, class_name: "Valet"
```
Comment (ggharry) [inline app/models/request.rb]: Do you not need a state field here?

File: config/environments/development.rb [hunk 4]
```diff
@@ -38,4 +38,19 @@
...
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
+  config.action_mailer.default_url_options = { :host => 'localhost:3000' }
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.raise_delivery_errors = false
+  config.action_mailer.default :charset => "utf-8"
+
+  config.action_mailer.smtp_settings = {
+  address: "smtp.gmail.com",
```
Comment (h8rry) [inline config/environments/development.rb]: Indentation

File: config/initializers/devise_token_auth.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  # example, using the default '/omniauth', the github oauth2 provider will
+  # redirect successful authentications to '/omniauth/github/callback'
+  #config.omniauth_prefix = "/omniauth"
+
+  # By defult sending current password is not needed for the password update.
+  # Uncomment to enforce current_password param to be checked before all
+  # attribute updates. Set it to :password if you want it to be checked only if
+  # password is updated.
+  # config.check_current_password_before_update = :attributes
+  config.default_confirm_success_url = "localhost:3000"
```
Comment (h8rry) [inline config/initializers/devise_token_auth.rb]: This should be an environment variable, for production.

File: db/migrate/20150908083456_create_requests.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+class CreateRequests < ActiveRecord::Migration
+  def change
+    create_table :requests do |t|
+      t.string :status
+      t.string :auth_code_pick_up
+      t.datetime :car_pick_up_time
+      t.string :bay_number
+      t.datetime :request_drop_off_time
+      t.string :auth_code_drop_off
+      t.integer :rating_pick_up
```
Comment (h8rry) [inline db/migrate/20150908083456_create_requests.rb:10]: These fields should have `_id` in the back

File: db/migrate/20150908084406_add_relationships_to_requests.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+class AddRelationshipsToRequests < ActiveRecord::Migration
+  def change
+    add_reference :requests, :user, references: :users, index: true
+    add_foreign_key :requests, :users
```
Comment (h8rry) [inline db/migrate/20150908084406_add_relationships_to_requests.rb]: You can replace these 2 lines with one line `add_reference :requests, :user, references: :users, index: true, foreign_key: true`

**Comment by h8rry** [comment]
The rest looks fine :)

File: db/migrate/20150908083456_create_requests.rb (hunk 6 above)
Comment (teelu) [inline db/migrate/20150908083456_create_requests.rb:10]: _id in the back?

**Comment by teelu** [comment]
What do you mean 
"These fields should have _id in the back" ?

File: db/migrate/20150908083456_create_requests.rb (hunk 6 above)
Comment (h8rry) [inline db/migrate/20150908083456_create_requests.rb:10]: `t.integer :rating_pick_up_id`

**Pull request merged**